### PR TITLE
Float -- episode 1 

### PR DESCRIPTION
**Pull request by shrit**
This PR allows to compile mnist_simple with `arma::fmat` instead of the standard `arma::mat`

@rcurtin feel free to review, I only tested mnist_simple. If necessary for the other will be a new episode of Float PRs

The duck is helpful with these complex errors, it took much less time than expected this time.

### CODE REVIEW

File: src/mlpack/methods/ann/layer/batch_norm_impl.hpp [hunk 1]
```diff
@@ -216,26 +216,26 @@ void BatchNormType<MatType>::Forward(
     // Calculate mean and variance over all channels.
     MatType mean = arma::sum(arma::sum(inputTemp, 2), 0) / m;
     variance = arma::sum(arma::sum(arma::pow(
-        inputTemp.each_slice() - arma::repmat(mean,
+        inputTemp.each_slice() - arma::repmat<MatType>(mean,
```
Comment (conradsnicta) [inline src/mlpack/methods/ann/layer/batch_norm_impl.hpp:219]: Not sure why `arma::repmat()` would need to explicitly have the `<MatType>` template parameter specified.  The function is designed to automatically work with all Armadillo matrix types.

Comment (shrit) [reply src/mlpack/methods/ann/layer/batch_norm_impl.hpp:219]: I know, I was getting a weird error at some point and I had to fix two other things, I can removed it and see if this work back.

Comment (shrit) [reply src/mlpack/methods/ann/layer/batch_norm_impl.hpp]: Done @conradsnicta

Comment (zoq) [reply src/mlpack/methods/ann/layer/batch_norm_impl.hpp]: We might need it later to forward it to the correct backend (armadillo / bandicoot). But looking at the code, we have to make quite a few changes to make it work with both backends, so probably not needed to do it now.

**Review by rcurtin — APPROVED** [review]
Nice, looks good to me. :+1: I'm sure there will be more work we have to do for the bandicoot adaptation, but getting `arma::fmat` to work throughout is a great first step.

**Comment by conradsnicta** [comment]
For posterity, this is related to #290

**Pull request merged**